client/state/ui/layout-focus/middleware.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import {
+	LAYOUT_FOCUS_SET,
+	LAYOUT_NEXT_FOCUS_ACTIVATE,
+	LAYOUT_NEXT_FOCUS_SET,
+} from 'state/action-types';
+import { getCurrentLayoutFocus } from 'state/ui/selectors';
+import { setLayoutFocus } from 'state/ui/actions';
+import layoutFocus from 'lib/layout-focus';
+
+/**
+ * Temporary Redux middleware intended to replay dispatched Redux layout focus
+ * operations on the legacy `lib/layout-focus` module and sync legacy area
+ * changes to global state.
+ *
+ * @return {Function} Redux middleware
+ */
+export default ( { dispatch, getState } ) => {
+	layoutFocus.on( 'change', () => {
```

Comment:
There's a [couple of middlewares that are only registered in the `window` context](https://github.com/Automattic/wp-calypso/blob/e213bbcaf1f415f51d9c8466099cdcd0ba4fcb90/client/state/index.js#L82-L89). I intentionally chose not to include it there; the assumption being layout focus is not specific to client, though perhaps I'm wrong. In any case, we could include this middleware client-side only if need be if it's a temporary transition.